MessageCenter3/Controllers/UserGroupsController.cs

```diff
@@ -42,22 +42,43 @@ public ActionResult<List<UserGroup>> SearchByUserId(int userId)
             return _repository.GetUserGroupsByUserId(userId);
         }
 
+        [Route("[action]/{userId}")]
+        [HttpGet]
+        public ActionResult<List<UserGroup>> GetAvailableUserGroups(int userId)
+        {
+            return _repository.GetAvailableUserGroups(userId);
+        }
+
         // POST api/userGroups
         [HttpPost]
         public ActionResult<UserGroup> AddGroup(UserGroupInputModel newGroup)
         {
             UserGroup group = _repository.AddUserGroup(newGroup);
-            _repository.AddUserToGroup(group.Id, newGroup.CreatorId);
+
+            JoinGroupRequestModel joinData = new JoinGroupRequestModel
+            {
+                GroupId = group.Id,
+                UserId = newGroup.CreatorId
+            };
```

Comment:
I suggest using AutoMapper for such things.

Reply:
Let's discuss this on Monday.

Reply:
Added AutoMapper